src/common/tokenCache.ts

```diff
@@ -0,0 +1,78 @@
+export type TokenRecord = {
+  token: string;
+  expiration: number;
+};
+
+export interface TokenCache {
+  getCachedToken(clientId: string, secret: string): TokenRecord | null;
+  cacheToken(
```

Comment:
At least in JDBC we use host as a part of key. IMHO this is important because theoretically the same application can connect to 2 different environments: staging and production.